### PR TITLE
Issue 2864: Create a alternate link for every status

### PR DESCRIPTION
**Pull request by annando**
See issue https://github.com/friendica/friendica/issues/2864

Hopefully it is what @Gargron needs.

### CODE REVIEW

File: mod/display.php [hunk 1]
```diff
@@ -11,6 +11,17 @@ function display_init(App $a) {
 	$nick = (($a->argc > 1) ? $a->argv[1] : '');
 	$profiledata = array();
 
+	if ($a->argc == 3) {
+		if (substr($a->argv[2], -5) == '.atom') {
+			require_once('include/dfrn.php');
```
Comment (fabrixxm) [inline mod/display.php]: I would move this to the top of the file..

Comment (annando) [inline mod/display.php]: Do we have a general rule for that? This is some special case, mostly we don't need that include.

I was unsure where to place it, but of course I can place it on the top.

Comment (annando) [inline mod/display.php]: Moved

**Comment by annando** [comment]
@Gargron Is this okay? Or should I change the first ```<dfrn:owner>``` to ```<author>```? Or is it enough to have the ```<author>``` inside the entry?

```
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xmlns:thr="http://purl.org/syndication/thread/1.0" xmlns:at="http://purl.org/atompub/tombstones/1.0" xmlns:media="http://purl.org/syndication/atommedia" xmlns:dfrn="http://purl.org/macgirvin/dfrn/1.0" xmlns:activity="http://activitystrea.ms/spec/1.0/" xmlns:georss="http://www.georss.org/georss" xmlns:poco="http://portablecontacts.net/spec/1.0" xmlns:ostatus="http://ostatus.org/schema/1.0" xmlns:statusnet="http://status.net/schema/api/1/">
  <id>https://pirati.ca/profile/heluecht</id>
  <title>Michael Vogel</title>
  <generator uri="https://friendi.ca" version="3.5.3dev-1227">Friendica</generator>
  <link rel="license" href="http://creativecommons.org/licenses/by/3.0/"/>
  <link rel="alternate" type="text/html" href="https://pirati.ca/profile/heluecht"/>
  <link href="https://pirati.ca/pubsubhubbub" rel="hub"/>
  <link rel="salmon" href="https://pirati.ca/salmon/heluecht"/>
  <link rel="http://salmon-protocol.org/ns/salmon-replies" href="https://pirati.ca/salmon/heluecht"/>
  <link rel="http://salmon-protocol.org/ns/salmon-mention" href="https://pirati.ca/salmon/heluecht"/>
  <dfrn:account_type>0</dfrn:account_type>
  <updated>2017-06-07T08:49:29Z</updated>
  <dfrn:owner>
    <name dfrn:updated="2016-09-24T22:19:36Z">Michael Vogel</name>
    <uri dfrn:updated="2016-09-24T22:19:36Z">https://pirati.ca/profile/heluecht</uri>
    <dfrn:handle dfrn:updated="2016-09-24T22:19:36Z">heluecht@pirati.ca</dfrn:handle>
    <link rel="photo" type="image/jpeg" media:width="175" media:height="175" href="https://pirati.ca/photo/07c4f9fc55f29d975947f35efda4aa7c-4.png" dfrn:updated="2016-09-16T14:02:08Z"/>
    <link rel="avatar" type="image/jpeg" media:width="175" media:height="175" href="https://pirati.ca/photo/07c4f9fc55f29d975947f35efda4aa7c-4.png" dfrn:updated="2016-09-16T14:02:08Z"/>
  </dfrn:owner>
  <entry>
    <author>
      <name>Michael Vogel</name>
      <uri>https://pirati.ca/profile/heluecht</uri>
      <dfrn:handle>heluecht@pirati.ca</dfrn:handle>
      <link rel="photo" type="image/jpeg" media:width="80" media:height="80" href="https://pirati.ca/photo/07c4f9fc55f29d975947f35efda4aa7c-4.png"/>
      <link rel="avatar" type="image/jpeg" media:width="80" media:height="80" href="https://pirati.ca/photo/07c4f9fc55f29d975947f35efda4aa7c-4.png"/>
    </author>
    <dfrn:owner>
      <name>Michael Vogel</name>
      <uri>https://pirati.ca/profile/heluecht</uri>
      <dfrn:handle>heluecht@pirati.ca</dfrn:handle>
      <link rel="photo" type="image/jpeg" media:width="80" media:height="80" href="https://pirati.ca/photo/07c4f9fc55f29d975947f35efda4aa7c-4.png"/>
      <link rel="avatar" type="image/jpeg" media:width="80" media:height="80" href="https://pirati.ca/photo/07c4f9fc55f29d975947f35efda4aa7c-4.png"/>
    </dfrn:owner>
    <ostatus:conversation href="https://pirati.ca/display/heluecht/15710456" ref="https://pirati.ca/display/heluecht/15710456">https://pirati.ca/display/heluecht/15710456</ostatus:conversation>
    <id>urn:X-dfrn:pirati.ca:1:ec054ce758593701e376f64659069209</id>
    <title/>
    <published>2017-06-06T19:26:27Z</published>
    <updated>2017-06-06T19:26:27Z</updated>
    <dfrn:env>VmVya2VocnNzY2hpbGRrdW5zdAoKW3VybD1odHRwczovL3BpcmF0aS5jYS9waG90b3MvaGVsdWVjaHQvaW1hZ2UvZjRlNTVlZDNmYmE2YjFkYmJlNzE3ZTNlYTgxZTE5NzFdW2ltZ11odHRwczovL3BpcmF0aS5jYS9waG90by9mNGU1NWVkM2ZiYTZiMWRiYmU3MTdlM2VhODFlMTk3MS0xLmpwZ1svaW1nXVsvdXJsXQ</dfrn:env>
    <content type="html">Verkehrsschildkunst&lt;br&gt;&lt;br&gt;&lt;img src="https://pirati.ca/photo/f4e55ed3fba6b1dbbe717e3ea81e1971-1.jpg" alt="Bild/Foto"&gt;&lt;a href="https://pirati.ca/photos/heluecht/image/f4e55ed3fba6b1dbbe717e3ea81e1971" class="attachment" rel="nofollow noopener" target="_blank"&gt;&lt;span class="invisible"&gt;https://&lt;/span&gt;&lt;span class="ellipsis"&gt;pirati.ca/photos/heluecht/imag&lt;/span&gt;&lt;span class="invisible"&gt;e/f4e55ed3fba6b1dbbe717e3ea81e1971&lt;/span&gt;&lt;/a&gt;</content>
    <link rel="alternate" type="text/html" href="https://pirati.ca/display/ec054ce758593701e376f64659069209"/>
    <dfrn:comment-allow>1</dfrn:comment-allow>
    <dfrn:location>Toto &amp; Malta, Johannes-Brahms-Platz 11, 20355 Hamburg, Deutschland</dfrn:location>
    <georss:point>53.55522445 9.98010864</georss:point>
    <statusnet:notice_info local_id="15710456" source="Twidere"/>
    <dfrn:diaspora_guid>ec054ce758593701e376f64659069209</dfrn:diaspora_guid>
    <activity:verb>http://activitystrea.ms/schema/1.0/post</activity:verb>
    <activity:object-type>http://activitystrea.ms/schema/1.0/image</activity:object-type>
  </entry>
</feed>
```

**Comment by Gargron** [comment]
Mastodon takes `dfrn:owner` when `author` is not present, iirc. That being said, if author inside all the entries is the same, it should be moved out to the feed for brevity sake

**Comment by annando** [comment]
The DFRN protocol can transport comments as well. For this it is important to have owner and author in the different entries as well.

So if you think you can work with this feed content then I would like to keep it that way.

**Comment by rabuzarus** [comment]
@annando 

wouldn't it be better to work with the item guid as "alternate" link instead of the item id? The item guid is unique and more more reliable. We also use the item guid in the url path in the display module. And the guid is part of the id in the created atom. As far as I know we use "direct" ID's only for internal use but not for the outside.

So if there is no downside I would suggest to use the guid to create an "alternate" link (guid.atom).

**Comment by annando** [comment]
There is a downside. Although it is likely impossible - but a guid could end with ".atom" - and that would create problems.

In old times we had exactly the format ```host.tld/display/nickname/id``` for the display links.

**Pull request merged**